<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Sparse Marginal Epistasis Test • smer</title>
<script src="lightswitch.js"></script><script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Sparse Marginal Epistasis Test">
<meta name="description" content="The Sparse Marginal Epistasis Test is a computationally efficient genetics method which detects statistical epistasis in complex traits. Stamp et al. (2025) &lt;doi:10.1101/2025.01.11.632557&gt;">
<meta property="og:description" content="The Sparse Marginal Epistasis Test is a computationally efficient genetics method which detects statistical epistasis in complex traits. Stamp et al. (2025) &lt;doi:10.1101/2025.01.11.632557&gt;">
<meta property="og:image" content="https://lcrawlab.github.io/sme/logo.png">
<script defer data-domain="pkgdown.r-lib.org,all.tidyverse.org" src="https://plausible.io/js/plausible.js"></script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">smer</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/smer.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Tutorials</h6></li>
    <li><a class="dropdown-item" href="articles/tutorial-cite-us.html">How To Cite Our Work</a></li>
    <li><a class="dropdown-item" href="articles/tutorial-create-mask-file.html">How To Create a Mask File</a></li>
    <li><a class="dropdown-item" href="articles/tutorial-memory-optimization.html">How To Optimize the Memory Requirements of SME</a></li>
    <li><a class="dropdown-item" href="articles/tutorial-simulate-traits.html">How To Simulate Traits</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Studies</h6></li>
    <li><a class="dropdown-item" href="articles/study-erythroid-differentiation-data.html">Conditioning Epistasis Search on Open Chromatin</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/lcrawlab/sme/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="the-sparse-marginal-epistasis-test-">The Sparse Marginal Epistasis test <a class="anchor" aria-label="anchor" href="#the-sparse-marginal-epistasis-test-"></a>
</h1>
</div>
<!-- badges: start -->
<!-- badges: end -->
<p>The <code>smer</code> package implements a computationally and statistically efficient method for detecting marginal epistasis in genome-wide association studies (GWAS). Find the full package documentation including examples and articles here: <a href="https://lcrawlab.github.io/sme/">Sparse Marginal Epistasis test Documentation</a>.</p>
<div class="section level2">
<h2 id="key-features">Key Features<a class="anchor" aria-label="anchor" href="#key-features"></a>
</h2>
<ul>
<li>Hutchinson’s stochastic trace estimator: efficient and scalable computation</li>
<li>Mailman algorithm: fast vector-by-matrix operation</li>
<li>Linear mixed model: controls for population structure</li>
<li>Multimodal Input: incorporates additional data from HDF5 files to improve power in detecting gene-by-gene interactions.</li>
<li>Optimize for Memory Constraints: Highly configurable block wise processing of the data allows to make the most of available resources. See also <a href="articles/tutorial-memory-optimization.html">How To Optimize the Memory Requirements of SME</a>.</li>
<li>Parallelization: Utilizes OpenMP for multi-threaded processing.</li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of <code>smer</code> from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"lcrawlab/sme"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="dependencies">Dependencies<a class="anchor" aria-label="anchor" href="#dependencies"></a>
</h2>
<p>System requirements of the package:</p>
<ul>
<li>GNU make</li>
<li>R (&gt;= 4.4)</li>
<li>Rhdf5lib (from BioConductor)</li>
<li>OpenMP (optional)</li>
</ul>
<p>To install <code>Rhdf5lib</code>, first install the tool <code>BiocManager</code> from CRAN, then install the library using this tool.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"Rhdf5lib"</span><span class="op">)</span></span></code></pre></div>
<p>The full list of R dependencies can be found in the <a href="https://github.com/lcrawlab/sme/blob/main/DESCRIPTION" class="external-link">DESCRIPTION file</a>.</p>
<div class="section level3">
<h3 id="openmp">OpenMP<a class="anchor" aria-label="anchor" href="#openmp"></a>
</h3>
<p>For OS X and Linux, the OpenMP library can be installed via one of the (shell) commands specified below:</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th align="left">System</th>
<th align="left">Command</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><strong>OS X (using Homebrew)</strong></td>
<td align="left"><code>brew install libomp</code></td>
</tr>
<tr class="even">
<td align="left"><strong>Debian-based systems (including Ubuntu)</strong></td>
<td align="left"><code>sudo apt-get install libomp-dev</code></td>
</tr>
</tbody>
</table>
<p>To enable openMP, it may be necessary to configure the compiler flags <code>SHLIB_OPENMP_CXXFLAGS</code> and <code>LDFLAGS</code> in the <code>~/.R/Makevars</code> file.</p>
<table class="table">
<thead><tr class="header">
<th>System</th>
<th>Required Flags</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>OS X</td>
<td><code>-Xclang -fopenmp -lomp</code></td>
</tr>
<tr class="even">
<td>Linux</td>
<td><code>-fopenmp -lomp</code></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="known-issues">Known Issues<a class="anchor" aria-label="anchor" href="#known-issues"></a>
</h2>
<p>Compiling the package requires the compiler to find the libraries for the dependencies. For unix systems, the libraries are typically installed at <code>/usr/local/lib</code> and <code>/usr/local/include</code>. For users using OS X and homebrew, the libraries are typically installed at <code>/opt/homebrew/lib</code> and <code>/opt/homebrew/include</code>.</p>
<p>Non-standard library paths need to be configured. The <code>src/Makevars</code> file configures the compiler flags and considers the <code>LDFLAGS</code> and <code>CPPFLAGS</code> from the <code>~/.R/Makevars</code> file.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Stamp J, Crawford L (2025). SME: The Sparse Marginal Epistasis Test. R package version 0.0.1, <a href="https://lcrawlab.github.io/sme/" class="uri">https://lcrawlab.github.io/sme/</a>, <a href="https://github.com/lcrawlab/sme" class="external-link uri">https://github.com/lcrawlab/sme</a>.</li>
<li>Stamp J, Smith Pattillo S, Weinreich D, Crawford L (2025). Sparse modeling of interactions enables fast detection of genome-wide epistasis in biobank-scale studies. biorxiv, <a href="https://doi.org/10.1101/2025.01.11.632557" class="external-link uri">https://doi.org/10.1101/2025.01.11.632557</a>
</li>
<li>Stamp J, Crawford L (2024). mvMAPIT: Multivariate Genome Wide Marginal Epistasis Test. R package version 2.0.3, <a href="https://lcrawlab.github.io/mvMAPIT/" class="external-link uri">https://lcrawlab.github.io/mvMAPIT/</a>, <a href="https://github.com/lcrawlab/mvMAPIT" class="external-link uri">https://github.com/lcrawlab/mvMAPIT</a>.</li>
<li>Stamp et al. (2023): Leveraging genetic correlation between traits for epistasis detection in GWAS. G3: Genes, Genomes, Genetics.</li>
<li>Fu, B., Pazokitoroudi, A., Xue, A., Anand, A., Anand, P., Zaitlen, N., &amp; Sankararaman, S. (2023). A biobank-scale test of marginal epistasis reveals genome-wide signals of polygenic epistasis. bioRxiv.</li>
<li>Crawford et al. (2017): Detecting epistasis with the marginal epistasis test. PLoS Genetics.</li>
<li>Devresse et al. (2024): HighFive - Header-only C++ HDF5 interface. <a href="https://zenodo.org/records/13120799" class="external-link uri">https://zenodo.org/records/13120799</a>
</li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lcrawlab/sme/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/lcrawlab/sme/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing smer</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Julian Stamp <br><small class="roles"> Maintainer, author </small> <a href="https://orcid.org/0000-0003-3014-6249" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Lorin Crawford <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-0178-8242" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Julian Stamp, Lorin Crawford.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
