<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Simulate Quantitative Traits from PLINK Genotypes — simulate_traits • smer</title><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Simulate Quantitative Traits from PLINK Genotypes — simulate_traits"><meta name="description" content="This function simulates a quantitative trait based on additive and epistatic
genetic effects using genotype data from a PLINK dataset. The simulated trait
is saved to a specified output file in a phenotype format compatible with
PLINK."><meta property="og:description" content="This function simulates a quantitative trait based on additive and epistatic
genetic effects using genotype data from a PLINK dataset. The simulated trait
is saved to a specified output file in a phenotype format compatible with
PLINK."><meta property="og:image" content="https://lcrawlab.github.io/sme/logo.png"><script defer data-domain="pkgdown.r-lib.org,all.tidyverse.org" src="https://plausible.io/js/plausible.js"></script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">smer</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/smer.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Tutorials</h6></li>
    <li><a class="dropdown-item" href="../articles/tutorial-cite-us.html">How To Cite Our Work</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-create-mask-file.html">How To Create a Mask File</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-memory-optimization.html">How To Optimize the Memory Requirements of SME</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-simulate-traits.html">How To Simulate Traits</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Studies</h6></li>
    <li><a class="dropdown-item" href="../articles/study-erythroid-differentiation-data.html">Conditioning Epistasis Search on Open Chromatin</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/lcrawlab/sme/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch"><li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Simulate Quantitative Traits from PLINK Genotypes</h1>
      <small class="dont-index">Source: <a href="https://github.com/lcrawlab/sme/blob/main/R/simulate_traits.R" class="external-link"><code>R/simulate_traits.R</code></a></small>
      <div class="d-none name"><code>simulate_traits.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function simulates a quantitative trait based on additive and epistatic
genetic effects using genotype data from a PLINK dataset. The simulated trait
is saved to a specified output file in a phenotype format compatible with
PLINK.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">simulate_traits</span><span class="op">(</span></span>
<span>  <span class="va">plink_file</span>,</span>
<span>  <span class="va">output_file</span>,</span>
<span>  <span class="va">additive_heritability</span>,</span>
<span>  <span class="va">gxg_heritability</span>,</span>
<span>  <span class="va">additive_indices</span>,</span>
<span>  <span class="va">gxg_indices_1</span>,</span>
<span>  <span class="va">gxg_indices_2</span>,</span>
<span>  log_level <span class="op">=</span> <span class="st">"WARNING"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-plink-file">plink_file<a class="anchor" aria-label="anchor" href="#arg-plink-file"></a></dt>
<dd><p>Character. Path to the PLINK dataset (without file
extension). The function will append <code>.bed</code>, <code>.bim</code>, and <code>.fam</code> extensions
as needed.</p></dd>


<dt id="arg-output-file">output_file<a class="anchor" aria-label="anchor" href="#arg-output-file"></a></dt>
<dd><p>Character. Path to the output file where the simulated
trait will be saved.</p></dd>


<dt id="arg-additive-heritability">additive_heritability<a class="anchor" aria-label="anchor" href="#arg-additive-heritability"></a></dt>
<dd><p>Numeric. A value between 0 and 1 specifying the
proportion of trait variance due to additive genetic effects.</p></dd>


<dt id="arg-gxg-heritability">gxg_heritability<a class="anchor" aria-label="anchor" href="#arg-gxg-heritability"></a></dt>
<dd><p>Numeric. A value between 0 and 1 specifying the
proportion of trait variance due to gene-by-gene (epistatic) interactions.
The sum of <code>additive_heritability</code> and <code>gxg_heritability</code> must not exceed 1.</p></dd>


<dt id="arg-additive-indices">additive_indices<a class="anchor" aria-label="anchor" href="#arg-additive-indices"></a></dt>
<dd><p>Integer vector. Indices of SNPs contributing to
additive genetic effects.</p></dd>


<dt id="arg-gxg-indices-">gxg_indices_1<a class="anchor" aria-label="anchor" href="#arg-gxg-indices-"></a></dt>
<dd><p>Integer vector. Indices of SNPs in the first group for
epistatic interactions.</p></dd>


<dt id="arg-gxg-indices-">gxg_indices_2<a class="anchor" aria-label="anchor" href="#arg-gxg-indices-"></a></dt>
<dd><p>Integer vector. Indices of SNPs in the second group for
epistatic interactions.</p></dd>


<dt id="arg-log-level">log_level<a class="anchor" aria-label="anchor" href="#arg-log-level"></a></dt>
<dd><p>Character. Logging level for messages
(e.g., "DEBUG", "INFO", "WARNING"). Default is "WARNING".</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>None. The simulated trait is written to the specified <code>output_file</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The function uses the following components to simulate the trait:</p><ul><li><p>Additive genetic effects: Determined by <code>additive_indices</code> and the
specified <code>additive_heritability</code>.</p></li>
<li><p>Epistatic interactions: Simulated using pairs of SNPs from <code>gxg_indices_1</code>
and <code>gxg_indices_2</code>, contributing to the <code>gxg_heritability</code>.</p></li>
<li><p>Environmental effects: Any remaining variance not explained by genetic
effects is assigned to random environmental noise.</p></li>
</ul><p>The output file is in PLINK-compatible phenotype format with three columns:
Family ID (<code>FID</code>), Individual ID (<code>IID</code>), and the simulated trait (<code>TRAIT</code>).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">plink_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\.bed"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, <span class="st">"test.bed"</span>, package <span class="op">=</span> <span class="st">"smer"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">out_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">additive_heritability</span> <span class="op">&lt;-</span> <span class="fl">0.3</span></span></span>
<span class="r-in"><span><span class="va">gxg_heritability</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span></span>
<span class="r-in"><span><span class="va">additive_snps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span>, <span class="fl">50</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gxg_group_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">additive_snps</span>, <span class="fl">10</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gxg_group_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">additive_snps</span>, <span class="va">gxg_group_1</span><span class="op">)</span>, <span class="fl">10</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">n_samples</span> <span class="op">&lt;-</span> <span class="fl">200</span></span></span>
<span class="r-in"><span><span class="fu">simulate_traits</span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="va">plink_file</span>,</span></span>
<span class="r-in"><span>  <span class="va">out_file</span>,</span></span>
<span class="r-in"><span>  <span class="va">additive_heritability</span>,</span></span>
<span class="r-in"><span>  <span class="va">gxg_heritability</span>,</span></span>
<span class="r-in"><span>  <span class="va">additive_snps</span>,</span></span>
<span class="r-in"><span>  <span class="va">gxg_group_1</span>,</span></span>
<span class="r-in"><span>  <span class="va">gxg_group_2</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">from_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">out_file</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">from_file</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   FID IID       TRAIT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   1   1 -0.65764347</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   2   1  0.46006105</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   3   1 -0.01990292</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   4   1 -0.86948835</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   5   1 -0.42641906</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   6   1  0.78883584</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Julian Stamp, Lorin Crawford.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

