<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>How To Use the Sparse Marginal Epistasis Test • smer</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="How To Use the Sparse Marginal Epistasis Test">
<script defer data-domain="pkgdown.r-lib.org,all.tidyverse.org" src="https://plausible.io/js/plausible.js"></script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">smer</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/smer.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Tutorials</h6></li>
    <li><a class="dropdown-item" href="../articles/tutorial-cite-us.html">How To Cite Our Work</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-create-mask-file.html">How To Create a Mask File</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-memory-optimization.html">How To Optimize the Memory Requirements of SME</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-simulate-traits.html">How To Simulate Traits</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Studies</h6></li>
    <li><a class="dropdown-item" href="../articles/study-erythroid-differentiation-data.html">Conditioning Epistasis Search on Open Chromatin</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/lcrawlab/sme/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>How To Use the Sparse Marginal Epistasis Test</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lcrawlab/sme/blob/main/vignettes/smer.Rmd" class="external-link"><code>vignettes/smer.Rmd</code></a></small>
      <div class="d-none name"><code>smer.Rmd</code></div>
    </div>

    
    
<p>The sparse marginal epistasis (SME) test performs a genome-wide
search for SNPs involved in genetic interactions while conditioning on
information derived from functional genomic data.</p>
<p>By examining one SNP at a time, SME fits a linear mixed model to test
for marginal epistasis. It explicitly models the combined additive
effects from all SNPs a marginal epistatic effect that represents
pairwise interactions involving a test SNP.</p>
<p>The key to the SME formulation is that the interaction between the
test SNP and other SNPs may be masked depending on additional
information. Masking interactions that do not contribute to trait
variance both maximizes the power of the inference as well as realizes
the computational efficiency needed to analyze human Biobank scale
data.</p>
<p>This pages presents a toy example to show case what insights the
implementation of the test provides.</p>
<p><img src="../reference/figures/sme-schematic-manhattan.png" width="100%" alt="Sparse Marginal Epistasis test (SME) schematic manhattan plot"><strong>Figure 1. </strong> SME performs a genome-wide search for SNPs
involved in genetic interactions while conditioning on information
derived from functional genomic data. SME has improved power to detect
marginal epistasis and runs 10x to 90x faster than state-of-the-art
methods.</p>
<div class="section level2">
<h2 id="run-sme-with-plink-data">Run SME with PLINK data<a class="anchor" aria-label="anchor" href="#run-sme-with-plink-data"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lcrawlab/sme" class="external-link">smer</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="data-requirements-and-file-formats">Data Requirements and File Formats<a class="anchor" aria-label="anchor" href="#data-requirements-and-file-formats"></a>
</h3>
<p>SME is implemented in R and requires your genetic data to be in PLINK
format, which consists of three files:</p>
<ul>
<li>
<code>.bim</code>: Contains SNP information</li>
<li>
<code>.bed</code>: Contains genetic data</li>
<li>
<code>.fam</code>: Contains sample information</li>
</ul>
<p><strong>Note:</strong> <code><a href="../reference/sme.html">sme()</a></code> cannot handle missing
genotypes. Prior to running <code><a href="../reference/sme.html">sme()</a></code>, use PLINK to remove
variants with missing genotypes or impute them.</p>
<p>Additionally, your phenotype (trait) data should be in a separate
file following PLINK’s phenotype format.</p>
<p>The <code><a href="../reference/sme.html">sme()</a></code> function includes parameters that let you
control memory usage and computational resources. For detailed guidance
on optimizing these settings for your system, please see our tutorial <a href="https://lcrawlab.github.io/sme/articles/tutorial-memory-optimization.html">How
To Optimize the Memory Requirements of SME</a>.</p>
</div>
<div class="section level3">
<h3 id="specifying-snps-for-analysis">Specifying SNPs for Analysis<a class="anchor" aria-label="anchor" href="#specifying-snps-for-analysis"></a>
</h3>
<p>When selecting which SNPs to analyze, you must provide their
positions as <strong>1-based indices</strong>. These indices correspond
to the row numbers in your <code>.bim</code> file, where the first SNP
is index 1, the second is index 2, and so on.</p>
<p>For complete details about all function parameters, please refer to
the documentation of the <code><a href="../reference/sme.html">sme()</a></code> function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># File inputs</span></span>
<span><span class="va">plink_file</span> <span class="op">&lt;-</span> <span class="st">"path/to/plink/file"</span></span>
<span><span class="va">pheno_file</span> <span class="op">&lt;-</span> <span class="st">"path/to/pheno/file"</span></span>
<span><span class="va">mask_file</span> <span class="op">&lt;-</span> <span class="st">"path/to/mask/file"</span></span>
<span></span>
<span><span class="co"># Parameter inputs</span></span>
<span><span class="va">chun_ksize</span> <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">n_randvecs</span> <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">n_blocks</span> <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">n_threads</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span></span>
<span><span class="co"># 1-based Indices of SNPs to be analyzed</span></span>
<span><span class="va">n_snps</span> <span class="op">&lt;-</span> <span class="fl">100</span></span>
<span><span class="va">snp_indices</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="va">n_snps</span></span>
<span></span>
<span><span class="va">sme_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sme.html">sme</a></span><span class="op">(</span></span>
<span>  <span class="va">plink_file</span>,</span>
<span>  <span class="va">pheno_file</span>,</span>
<span>  <span class="va">mask_file</span>,</span>
<span>  <span class="va">snp_indices</span>,</span>
<span>  <span class="va">chunk_size</span>,</span>
<span>  <span class="va">n_randvecs</span>,</span>
<span>  <span class="va">n_blocks</span>,</span>
<span>  <span class="va">n_threads</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>This analysis uses simulated data for demonstration purposes. We
simulated synthetic phenotypes from 5000 synthetic genotypes with 6000
SNPs. If you would like to learn how to simulate data, please refer to
our tutorial <a href="https://lcrawlab.github.io/sme/articles/tutorial-simulate-traits.html">How
To Simulate Traits</a>.</p>
</div>
<div class="section level3">
<h3 id="understanding-the-results">Understanding the Results<a class="anchor" aria-label="anchor" href="#understanding-the-results"></a>
</h3>
<p>When you call the <code><a href="../reference/sme.html">sme()</a></code> function, it returns a list
containing multiple elements. The most important one is called
<code>summary</code>, which contains the main analysis results. These
results are formatted as tidy data, making them compatible with popular
R packages like <code>ggplot2</code> and <code>dplyr</code> for further
analysis and visualization.</p>
</div>
<div class="section level3">
<h3 id="visualizing-genomic-associations">Visualizing Genomic Associations<a class="anchor" aria-label="anchor" href="#visualizing-genomic-associations"></a>
</h3>
<p>We use Manhattan plots to visualize genome-wide analyses because they
effectively highlight strong associations between genetic variants and
traits. In this case, the Manhattan plot specifically shows statistical
epistasis (interactions between genes). For reference, we’ve marked the
true causal SNPs (Single Nucleotide Polymorphisms) in green on the plot
- these are the genetic variants we included in our simulation as having
real effects.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sme_result</span><span class="op">$</span><span class="va">summary</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">index</span>,</span>
<span>  y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>,</span>
<span>  color <span class="op">=</span> <span class="va">true_gxg_snp</span></span>
<span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Position"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"Epistatic SNP"</span><span class="op">)</span></span></code></pre></div>
<p><img src="smer_files/figure-html/manhattan_plot-1.png" alt="Manhattan plot to illustrate the sparse marginal epistasis test" width="700"></p>
</div>
<div class="section level3">
<h3 id="understanding-variance-components-and-effect-sizes">Understanding Variance Components and Effect Sizes<a class="anchor" aria-label="anchor" href="#understanding-variance-components-and-effect-sizes"></a>
</h3>
<p>SME estimates how much of the total trait variation can be explained
by genetic interactions. In this simulation, we set the Phenotypic
Variance Explained (PVE) to 5% for each SNP involved in epistatic
interactions.</p>
<p>The plot below shows how well our method recovered these effects. It
displays two distributions:</p>
<ol style="list-style-type: decimal">
<li>The estimated PVE for SNPs we know are truly involved in epistatic
interactions</li>
<li>The estimated PVE for SNPs that have no real epistatic effects</li>
</ol>
<p>The dashed line marks the true 5% PVE level we used in the
simulation, allowing you to see how accurately the method estimated the
actual effect sizes.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sme_result</span><span class="op">$</span><span class="va">summary</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">true_gxg_snp</span>, y <span class="op">=</span> <span class="va">pve</span>, fill <span class="op">=</span> <span class="va">true_gxg_snp</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0.05</span>, color <span class="op">=</span> <span class="st">"grey40"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html" class="external-link">annotate</a></span><span class="op">(</span><span class="st">"text"</span>, x <span class="op">=</span> <span class="fl">0.8</span>, y <span class="op">=</span>  <span class="fl">0.055</span>,</span>
<span>           label <span class="op">=</span> <span class="st">"True per SNP epistatic PVE"</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Epistatic SNP"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Phenotypic Variance Explained"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code></pre></div>
<p><img src="smer_files/figure-html/pve_plot-1.png" alt="PVE plot to illustrate the sparse marginal epistasis test" width="700"></p>
</div>
<div class="section level3">
<h3 id="narrow-sense-heritability-estimates">Narrow Sense Heritability Estimates<a class="anchor" aria-label="anchor" href="#narrow-sense-heritability-estimates"></a>
</h3>
<p>The SME method uses a linear mixed model to separate different
sources of trait variation. One key component it estimates is narrow
sense heritability
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>h</mi><mn>2</mn></msup><annotation encoding="application/x-tex">h^2</annotation></semantics></math>),
which measures how much trait variation can be explained by additive
genetic effects.</p>
<p>The plot below breaks down the estimated sources of trait
variation:</p>
<ul>
<li>“grm”: Shows narrow sense heritability
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>h</mi><mn>2</mn></msup><annotation encoding="application/x-tex">h^2</annotation></semantics></math>)</li>
<li>“gxg”: Shows variance due to gene-by-gene interactions</li>
<li>“error”: Shows unexplained variance, such as environmental
effects</li>
</ul>
<p>In our simulation, we set the true narrow sense heritability to 30%,
shown as a dashed line in the plot. This reference line helps you
evaluate how accurately SME estimated the genetic components of trait
variation.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sme_result</span><span class="op">$</span><span class="va">vc_estimate</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">component</span>, y <span class="op">=</span> <span class="va">vc_estimate</span>, fill <span class="op">=</span> <span class="va">component</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0.3</span>, color <span class="op">=</span> <span class="st">"grey40"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html" class="external-link">annotate</a></span><span class="op">(</span><span class="st">"text"</span>, x <span class="op">=</span> <span class="fl">0.7</span>, y <span class="op">=</span>  <span class="fl">0.33</span>,</span>
<span>           label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="st">"True "</span> <span class="op">*</span> <span class="va">h</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>  <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Component"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Variance Component Estimate"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code></pre></div>
<p><img src="smer_files/figure-html/h2_plot-1.png" alt="h2 plot to illustrate the sparse marginal epistasis test" width="700"></p>
<p>The estimate of the narrow-sense heritability
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>h</mi><mn>2</mn></msup><annotation encoding="application/x-tex">h^2</annotation></semantics></math>
is much less variable because it is always informed by the same genetic
relatedness matrix. In this small data example it overestimates the
heritability but it is unbiased in general.</p>
</div>
</div>
<div class="section level2">
<h2 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ggplot2_3.5.1 dplyr_1.1.4   smer_0.0.1   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] gtable_0.3.6        jsonlite_1.8.9      compiler_4.4.2     </span></span>
<span><span class="co">#&gt;  [4] tidyselect_1.2.1    Rcpp_1.0.14         FMStable_0.1-4     </span></span>
<span><span class="co">#&gt;  [7] parallel_4.4.2      tidyr_1.3.1         jquerylib_0.1.4    </span></span>
<span><span class="co">#&gt; [10] scales_1.3.0        systemfonts_1.1.0   textshaping_0.4.1  </span></span>
<span><span class="co">#&gt; [13] harmonicmeanp_3.0.1 yaml_2.3.10         fastmap_1.2.0      </span></span>
<span><span class="co">#&gt; [16] R6_2.5.1            labeling_0.4.3      generics_0.1.3     </span></span>
<span><span class="co">#&gt; [19] knitr_1.49          genio_1.1.2         iterators_1.0.14   </span></span>
<span><span class="co">#&gt; [22] backports_1.5.0     checkmate_2.3.2     tibble_3.2.1       </span></span>
<span><span class="co">#&gt; [25] desc_1.4.3          munsell_0.5.1       bslib_0.8.0        </span></span>
<span><span class="co">#&gt; [28] pillar_1.10.1       rlang_1.1.4         cachem_1.1.0       </span></span>
<span><span class="co">#&gt; [31] xfun_0.50           fs_1.6.5            sass_0.4.9         </span></span>
<span><span class="co">#&gt; [34] cli_3.6.3           withr_3.0.2         pkgdown_2.1.1      </span></span>
<span><span class="co">#&gt; [37] magrittr_2.0.3      grid_4.4.2          digest_0.6.37      </span></span>
<span><span class="co">#&gt; [40] mvMAPIT_2.0.3       foreach_1.5.2       mvtnorm_1.3-3      </span></span>
<span><span class="co">#&gt; [43] lifecycle_1.0.4     CompQuadForm_1.4.3  vctrs_0.6.5        </span></span>
<span><span class="co">#&gt; [46] evaluate_1.0.3      glue_1.8.0          farver_2.1.2       </span></span>
<span><span class="co">#&gt; [49] codetools_0.2-20    ragg_1.3.3          colorspace_2.1-1   </span></span>
<span><span class="co">#&gt; [52] purrr_1.0.2         rmarkdown_2.29      tools_4.4.2        </span></span>
<span><span class="co">#&gt; [55] pkgconfig_2.0.3     htmltools_0.5.8.1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Julian Stamp, Lorin Crawford.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
