<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Conditioning Epistasis Search on Open Chromatin • smer</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Conditioning Epistasis Search on Open Chromatin">
<script defer data-domain="pkgdown.r-lib.org,all.tidyverse.org" src="https://plausible.io/js/plausible.js"></script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">smer</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/smer.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Tutorials</h6></li>
    <li><a class="dropdown-item" href="../articles/tutorial-cite-us.html">How To Cite Our Work</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-create-mask-file.html">How To Create a Mask File</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-memory-optimization.html">How To Optimize the Memory Requirements of SME</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-simulate-traits.html">How To Simulate Traits</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Studies</h6></li>
    <li><a class="dropdown-item" href="../articles/study-erythroid-differentiation-data.html">Conditioning Epistasis Search on Open Chromatin</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/lcrawlab/sme/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Conditioning Epistasis Search on Open Chromatin</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lcrawlab/sme/blob/main/vignettes/study-erythroid-differentiation-data.Rmd" class="external-link"><code>vignettes/study-erythroid-differentiation-data.Rmd</code></a></small>
      <div class="d-none name"><code>study-erythroid-differentiation-data.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lcrawlab/sme" class="external-link">smer</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="dnase-i-hypersensitive-sites-of-erythroid-differentiation-reveal-statistical-epistasis-in-human-hematology-traits">DNAse I hypersensitive sites of erythroid differentiation reveal
statistical epistasis in human hematology traits<a class="anchor" aria-label="anchor" href="#dnase-i-hypersensitive-sites-of-erythroid-differentiation-reveal-statistical-epistasis-in-human-hematology-traits"></a>
</h2>
<p>We apply SME to hematology traits of white British individuals from
the UK Biobank. After quality control, the remaining data are 349,411
individuals and 543,813 SNPs of common variants. We select the traits
mean corpuscular hemoglobin (MCH), mean corpuscular hemoglobin
concentration, mean corpuscular volume (MCV), and hematocrit (HCT). As
external sparse data source, we leverage DNase I-hypersensitive sites
(DHSs) data measured over 12 days of ex-vivo erythroid differentiation
(Georgolopoulos et al. 2024). DHS are enriched in transcriptional
activity and used to identify regulatory DNA.</p>
<p>The first three traits, MCH, MCHC, and MCV are traits of the red
blood cells (RBC). Previous GWAS studies found that genes associated
with these traits are implicated in erythroid differentiation.
Therefore, we expect that genomic data that indicates regulatory regions
gathered during erythropoiesis will be informative for these traits. HCT
measures the percentage of red blood cells in the blood. The maturation
of erythroid progenitor cells is regulated by an oxygen-sensing
mechanism. We hypothesise that HCT, is not informed by functional data
of erythropoiesis.</p>
<div class="section level3">
<h3 id="mask-file-preparation">Mask File Preparation<a class="anchor" aria-label="anchor" href="#mask-file-preparation"></a>
</h3>
<p>The external data sources used in this study represented genomic
intervals of DHS regions and LD blocks. In the following we mock this
data to illustrate how to create a mask file for <code><a href="../reference/sme.html">sme()</a></code>. See
<a href="https://lcrawlab.github.io/sme/articles/tutorial-create-mask-file.html">How
To Create a Mask File</a> for more details.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bim_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  chromosome <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>  variant_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rs1"</span>, <span class="st">"rs2"</span>, <span class="st">"rs3"</span>, <span class="st">"rs4"</span>, <span class="st">"rs5"</span>, <span class="st">"rs6"</span>, <span class="st">"rs7"</span>, <span class="st">"rs8"</span>, <span class="st">"rs9"</span><span class="op">)</span>,</span>
<span>  cm_position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>  bp_position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">20</span>, <span class="fl">30</span>, <span class="fl">40</span>, <span class="fl">50</span>, <span class="fl">60</span>, <span class="fl">70</span>, <span class="fl">80</span>, <span class="fl">90</span><span class="op">)</span>,</span>
<span>  allele1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"G"</span>, <span class="st">"C"</span>, <span class="st">"C"</span>, <span class="st">"T"</span>, <span class="st">"T"</span>, <span class="st">"A"</span><span class="op">)</span>,</span>
<span>  allele1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"G"</span>, <span class="st">"G"</span>, <span class="st">"G"</span>, <span class="st">"A"</span>, <span class="st">"T"</span>, <span class="st">"T"</span>, <span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"G"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">bim_data</span><span class="op">$</span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">bim_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># DHS intervals</span></span>
<span><span class="va">hg19_dhs_regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  chromosome <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>  start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">45</span>, <span class="fl">85</span><span class="op">)</span>,</span>
<span>  stop <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">15</span>, <span class="fl">55</span>, <span class="fl">95</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># LD block intervals</span></span>
<span><span class="va">hg19_ld_blocks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  chromosome <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>  start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">25</span>, <span class="fl">35</span>, <span class="fl">45</span>, <span class="fl">65</span>, <span class="fl">75</span>, <span class="fl">85</span><span class="op">)</span>,</span>
<span>  stop <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">25</span>, <span class="fl">35</span>, <span class="fl">45</span>, <span class="fl">65</span>, <span class="fl">75</span>, <span class="fl">85</span>, <span class="fl">95</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>We make use of the package <code>GenomicRanges</code> to efficiently
map the <code>PLINK</code> data to the intervals of the DHS and LD
data.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert .bim to GRanges object</span></span>
<span><span class="va">bim_gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span></span>
<span>  seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="va">bim_data</span><span class="op">$</span><span class="va">chromosome</span><span class="op">)</span>,</span>
<span>  ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="va">bim_data</span><span class="op">$</span><span class="va">bp_position</span>, end <span class="op">=</span> <span class="va">bim_data</span><span class="op">$</span><span class="va">bp_position</span><span class="op">)</span>,</span>
<span>  variant_id <span class="op">=</span> <span class="va">bim_data</span><span class="op">$</span><span class="va">variant_id</span>,</span>
<span>  genome <span class="op">=</span> <span class="st">"hg19"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert DHS to GRanges object</span></span>
<span><span class="va">dhs_gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span></span>
<span>  seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="va">hg19_dhs_regions</span><span class="op">$</span><span class="va">chromosome</span><span class="op">)</span>,</span>
<span>  ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="va">hg19_dhs_regions</span><span class="op">$</span><span class="va">start</span>, end <span class="op">=</span> <span class="va">hg19_dhs_regions</span><span class="op">$</span><span class="va">stop</span><span class="op">)</span>,</span>
<span>  genome <span class="op">=</span> <span class="st">"hg19"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Find overlaps of BIM variants and DHS intervals</span></span>
<span><span class="va">overlaps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span><span class="va">bim_gr</span>, <span class="va">dhs_gr</span>, maxgap <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Extract overlapping variants</span></span>
<span><span class="va">dhs_data</span> <span class="op">&lt;-</span> <span class="va">bim_data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">overlaps</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">dhs_data</span> <span class="op">&lt;-</span> <span class="va">dhs_data</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">dhs_data</span><span class="op">$</span><span class="va">index</span><span class="op">)</span>, <span class="op">]</span></span></code></pre></div>
<p>Of the 543,813 SNPs in our data, 4,932 are located in the DHS
regions. The DHS regions in this data are distributed along the whole
genome. To test for marginal epistasis with SME we consider the variants
in the DHS regions important.</p>
<p>Next we map the <code>PLINK</code> data to the LD blocks.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert to GRanges object</span></span>
<span><span class="va">ld_gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span></span>
<span>  seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="va">hg19_ld_blocks</span><span class="op">$</span><span class="va">chromosome</span><span class="op">)</span>,</span>
<span>  ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="va">hg19_ld_blocks</span><span class="op">$</span><span class="va">start</span>, end <span class="op">=</span> <span class="va">hg19_ld_blocks</span><span class="op">$</span><span class="va">stop</span><span class="op">)</span>,</span>
<span>  genome <span class="op">=</span> <span class="st">"hg19"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Find LD block of bim variants</span></span>
<span><span class="va">ld_overlaps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">bim_gr</span>, subject <span class="op">=</span> <span class="va">ld_gr</span><span class="op">)</span></span></code></pre></div>
<p>With these objects, we can create the mask file. With larger data, we
recommend splitting the <code>PLINK</code> variants that are analyzed
into batches, create one mask file per batch, and submit one job per
batch on a High Peformance Cluster.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">output_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">gxg_group</span> <span class="op">&lt;-</span> <span class="st">"gxg"</span></span>
<span><span class="va">ld_group</span> <span class="op">&lt;-</span> <span class="st">"ld"</span></span>
<span></span>
<span><span class="va">gxg_variants</span> <span class="op">&lt;-</span> <span class="va">dhs_data</span><span class="op">$</span><span class="va">index</span> <span class="op">-</span> <span class="fl">1</span> <span class="co"># 0-base index for C++</span></span>
<span></span>
<span><span class="fu"><a href="../reference/create_hdf5_file.html">create_hdf5_file</a></span><span class="op">(</span><span class="va">output_file</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="va">bim_data</span><span class="op">$</span><span class="va">index</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span> <span class="op">{</span> <span class="co"># 0-base index for C++</span></span>
<span>  <span class="co"># Write DHS mask</span></span>
<span>  <span class="va">gxg_ds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"%s/%d"</span>, <span class="va">gxg_group</span>, <span class="va">j</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="../reference/write_hdf5_dataset.html">write_hdf5_dataset</a></span><span class="op">(</span>file_name <span class="op">=</span> <span class="va">output_file</span>,</span>
<span>                     dataset_name <span class="op">=</span> <span class="va">gxg_ds</span>,</span>
<span>                     <span class="va">gxg_variants</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># Find LD block of focal SNP</span></span>
<span>  <span class="va">focal_gr</span> <span class="op">&lt;-</span> <span class="va">ld_gr</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">ld_overlaps</span><span class="op">[</span><span class="va">j</span>,<span class="op">]</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span>  <span class="co"># Find variants in LD block of focal SNP</span></span>
<span>  <span class="va">focal_ld</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">bim_gr</span>, subject <span class="op">=</span> <span class="va">focal_gr</span><span class="op">)</span></span>
<span>  <span class="va">ld_data</span> <span class="op">&lt;-</span> <span class="va">bim_data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">focal_ld</span><span class="op">)</span>,<span class="op">]</span></span>
<span>  <span class="va">ld_variants</span> <span class="op">&lt;-</span> <span class="va">ld_data</span><span class="op">$</span><span class="va">index</span> <span class="op">-</span> <span class="fl">1</span> <span class="co"># 0-base index for C++</span></span>
<span></span>
<span>  <span class="co"># Write LD mask</span></span>
<span>  <span class="va">ld_ds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"%s/%d"</span>, <span class="va">ld_group</span>, <span class="va">j</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="../reference/write_hdf5_dataset.html">write_hdf5_dataset</a></span><span class="op">(</span>file_name <span class="op">=</span> <span class="va">output_file</span>,</span>
<span>                     dataset_name <span class="op">=</span> <span class="va">ld_ds</span>,</span>
<span>                     <span class="va">ld_variants</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">dhs_indices</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_hdf5_dataset.html">read_hdf5_dataset</a></span><span class="op">(</span>file_name <span class="op">=</span> <span class="va">output_file</span>, dataset_name <span class="op">=</span> <span class="va">gxg_ds</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"DHS indices: %s"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">dhs_indices</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "DHS indices: 0, 4, 8"</span></span></code></pre></div>
<p>With this mask file we can run SME.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sme_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sme.html">sme</a></span><span class="op">(</span></span>
<span>  plink_file <span class="op">=</span> <span class="st">"/path/to/plink/data"</span>,</span>
<span>  pheno_file <span class="op">=</span> <span class="st">"/path/to/pheno/data"</span>,</span>
<span>  mask_file <span class="op">=</span> <span class="st">"/path/to/mask/file"</span>,</span>
<span>  gxg_indices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>  chunk_size <span class="op">=</span> <span class="fl">250</span>,</span>
<span>  n_randvecs <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  n_blocks <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  n_threads <span class="op">=</span> <span class="fl">6</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The genome-wide association test for marginal epistasis in the red
blood cell traits MCH and MCV finds genome-wide significant statistical
epistasis
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>&lt;</mo><mrow><mn>5</mn><mo> × </mo><msup><mn>10</mn><mn>−8</mn></msup></mrow></mrow><annotation encoding="application/x-tex">P
&lt; \num{5e-8}</annotation></semantics></math>) on chromosome 6 (Fig.
1). Importantly, most of the SNPs or the genes which they map to have
been previously discovered for non-additive gene action related to
erythropoiesis and RBC traits.</p>
</div>
</div>
<div class="section level2">
<h2 id="section">
<img src="../reference/figures/dhs_manhattan.png" width="100%" alt="Erythroid differentiation DHS sites SME analysis"><a class="anchor" aria-label="anchor" href="#section"></a>
</h2>
<p><strong>Figure 1.</strong> Manhattan plot of the SME analysis. The
dashed blue line is the significance threshold after Bonferroni
correction.</p>
<p>The strongest association in the trait MCH, SNP rs4711092
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mrow><mn>1.41</mn><mo> × </mo><msup><mn>10</mn><mn>−11</mn></msup></mrow></mrow><annotation encoding="application/x-tex">P =
\num{1.41e-11}</annotation></semantics></math>, PVE 0.7%), maps to the
gene secretagogin (). The gene regulates exocytosis by interacting with
two soluble NSF adaptor proteins ( and ) and is critical for cell growth
in some tissues. A total of five SNPs which SME significantly associates
with MCH (strongest association with rs9366624
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mrow><mn>1.8</mn><mo> × </mo><msup><mn>10</mn><mn>−9</mn></msup></mrow></mrow><annotation encoding="application/x-tex">P = \num{1.8e-9}</annotation></semantics></math>,
PVE 1.1%) are in the gene capping protein regulator and myosin 1 linker
1 (). The gene is known to interact with and regulate caping protein ().
plays a role via protein-protein interaction in regulating
erythrpoiesis. Specifically, proteins regulate actin dynamics by
regulating the activity of . Erythropoiesis leads to modifications in
the expression of membrane and cytoskeletal proteins, whose interactions
impact cell structure and function. Both genes and have previously been
associated with hemoglobin concentration. The strongest association in
the trait MCV, SNP rs9276
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mrow><mn>9.09</mn><mo> × </mo><msup><mn>10</mn><mn>−10</mn></msup></mrow></mrow><annotation encoding="application/x-tex">P = \num{9.09e-10}</annotation></semantics></math>,
PVE 0.24%) maps to the gene of the major histocompatibility complex.
With the SNP rs9366624
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mrow><mn>1.86</mn><mo> × </mo><msup><mn>10</mn><mn>−8</mn></msup></mrow></mrow><annotation encoding="application/x-tex">P = \num{1.86e-8}</annotation></semantics></math>,
PVE 0.8%), also the gene is significantly associated with the trait for
marginal epistasis. The complete list of significant associations
produced by SME is reported in Tab. 1.</p>
<table class="table">
<colgroup>
<col width="13%">
<col width="13%">
<col width="20%">
<col width="21%">
<col width="19%">
<col width="13%">
</colgroup>
<thead><tr class="header">
<th><strong>Trait</strong></th>
<th><strong>ID</strong></th>
<th><strong>Coordinates</strong></th>
<th><strong><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>P</mi><annotation encoding="application/x-tex">P</annotation></semantics></math>-Value</strong></th>
<th><strong>PVE</strong></th>
<th><strong>Gene</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td>MCH</td>
<td>rs4711092</td>
<td>chr6:25684405</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>1.41</mn><mo> × </mo><msup><mn>10</mn><mn>−11</mn></msup></mrow><annotation encoding="application/x-tex">\num{1.41e-11}</annotation></semantics></math></td>
<td>0.007 (0.0009)</td>
<td><em>SCGN</em></td>
</tr>
<tr class="even">
<td>MCH</td>
<td>rs9366624</td>
<td>chr6:25439492</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>1.8</mn><mo> × </mo><msup><mn>10</mn><mn>−9</mn></msup></mrow><annotation encoding="application/x-tex">\num{1.8e-9}</annotation></semantics></math></td>
<td>0.011 (0.002)</td>
<td><em>CARMIL1</em></td>
</tr>
<tr class="odd">
<td>MCH</td>
<td>rs9461167</td>
<td>chr6:25418571</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>2.34</mn><mo> × </mo><msup><mn>10</mn><mn>−9</mn></msup></mrow><annotation encoding="application/x-tex">\num{2.34e-9}</annotation></semantics></math></td>
<td>0.007 (0.001)</td>
<td><em>CARMIL1</em></td>
</tr>
<tr class="even">
<td>MCH</td>
<td>rs9379764</td>
<td>chr6:25414023</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>5.53</mn><mo> × </mo><msup><mn>10</mn><mn>−9</mn></msup></mrow><annotation encoding="application/x-tex">\num{5.53e-9}</annotation></semantics></math></td>
<td>0.012 (0.002)</td>
<td><em>CARMIL1</em></td>
</tr>
<tr class="odd">
<td>MCH</td>
<td>rs441460</td>
<td>chr6:25548288</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>1.2</mn><mo> × </mo><msup><mn>10</mn><mn>−8</mn></msup></mrow><annotation encoding="application/x-tex">\num{1.2e-8}</annotation></semantics></math></td>
<td>0.008 (0.001)</td>
<td><em>CARMIL1</em></td>
</tr>
<tr class="even">
<td>MCH</td>
<td>rs198834</td>
<td>chr6:26114372</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>2.77</mn><mo> × </mo><msup><mn>10</mn><mn>−8</mn></msup></mrow><annotation encoding="application/x-tex">\num{2.77e-8}</annotation></semantics></math></td>
<td>0.008 (0.001)</td>
<td><em>H2BC4</em></td>
</tr>
<tr class="odd">
<td>MCH</td>
<td>rs13203202</td>
<td>chr6:25582771</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>3.17</mn><mo> × </mo><msup><mn>10</mn><mn>−8</mn></msup></mrow><annotation encoding="application/x-tex">\num{3.17e-8}</annotation></semantics></math></td>
<td>0.012 (0.002)</td>
<td><em>CARMIL1</em></td>
</tr>
<tr class="even">
<td>MCV</td>
<td>rs9276</td>
<td>chr6:33053577</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>9.09</mn><mo> × </mo><msup><mn>10</mn><mn>−10</mn></msup></mrow><annotation encoding="application/x-tex">\num{9.09e-10}</annotation></semantics></math></td>
<td>0.0024 (0.0004)</td>
<td><em>HLA-DPB1</em></td>
</tr>
<tr class="odd">
<td>MCV</td>
<td>rs9366624</td>
<td>chr6:25439492</td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>1.86</mn><mo> × </mo><msup><mn>10</mn><mn>−8</mn></msup></mrow><annotation encoding="application/x-tex">\num{1.86e-8}</annotation></semantics></math></td>
<td>0.008 (0.001)</td>
<td><em>CARMIL1</em></td>
</tr>
</tbody>
</table>
<p><strong>Table 1.</strong> Significant trait associations for marginal
epistasis.</p>
<p>Fitting the linear mixed model of SME also produces narrow-sense
heritability estimates equivalent to RHE regression. The heritability
estimates of SME for the four traits in this study are similar to
heritability estimates found in the literature(Tab. 2).</p>
<table class="table">
<thead><tr class="header">
<th><strong>Trait</strong></th>
<th><strong><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>h</mi><mn>2</mn></msup><annotation encoding="application/x-tex">h^2</annotation></semantics></math></strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td>HCT</td>
<td>0.28 (0.01)</td>
</tr>
<tr class="even">
<td>MCH</td>
<td>0.56 (0.03)</td>
</tr>
<tr class="odd">
<td>MCHC</td>
<td>0.14 (0.01)</td>
</tr>
<tr class="even">
<td>MCV</td>
<td>0.52 (0.03)</td>
</tr>
</tbody>
</table>
<p><strong>Table 2.</strong> Narrow-sense heritability
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>h</mi><mn>2</mn></msup><annotation encoding="application/x-tex">h^2</annotation></semantics></math>)
estimates in the SME analysis.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Stamp J, Smith Pattillo S, Weinreich D, Crawford L (2025). Sparse
modeling of interactions enables fast detection of genome-wide epistasis
in biobank-scale studies. biorxiv, <a href="https://doi.org/10.1101/2025.01.11.632557" class="external-link uri">https://doi.org/10.1101/2025.01.11.632557</a>
</li>
<li>Georgolopoulos, G. GEO Data Set: Discrete regulatory modules
instruct hematopoietic lineage commitment and differentiation <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE182816" class="external-link uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE182816</a>
(2024).</li>
</ul>
</div>
<div class="section level2">
<h2 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] smer_0.0.1           GenomicRanges_1.60.0 GenomeInfoDb_1.44.0 </span></span>
<span><span class="co">#&gt; [4] IRanges_2.42.0       S4Vectors_0.46.0     BiocGenerics_0.54.0 </span></span>
<span><span class="co">#&gt; [7] generics_0.1.4      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] sass_0.4.10             tidyr_1.3.1             digest_0.6.37          </span></span>
<span><span class="co">#&gt;  [4] magrittr_2.0.3          evaluate_1.0.4          iterators_1.0.14       </span></span>
<span><span class="co">#&gt;  [7] CompQuadForm_1.4.4      mvtnorm_1.3-3           fastmap_1.2.0          </span></span>
<span><span class="co">#&gt; [10] foreach_1.5.2           genio_1.1.2             jsonlite_2.0.0         </span></span>
<span><span class="co">#&gt; [13] backports_1.5.0         httr_1.4.7              purrr_1.1.0            </span></span>
<span><span class="co">#&gt; [16] UCSC.utils_1.4.0        codetools_0.2-20        textshaping_1.0.1      </span></span>
<span><span class="co">#&gt; [19] jquerylib_0.1.4         cli_3.6.5               rlang_1.1.6            </span></span>
<span><span class="co">#&gt; [22] XVector_0.48.0          cachem_1.1.0            yaml_2.3.10            </span></span>
<span><span class="co">#&gt; [25] FMStable_0.1-4          tools_4.5.1             parallel_4.5.1         </span></span>
<span><span class="co">#&gt; [28] checkmate_2.3.2         dplyr_1.1.4             GenomeInfoDbData_1.2.14</span></span>
<span><span class="co">#&gt; [31] vctrs_0.6.5             R6_2.6.1                lifecycle_1.0.4        </span></span>
<span><span class="co">#&gt; [34] mvMAPIT_2.0.3           fs_1.6.6                ragg_1.4.0             </span></span>
<span><span class="co">#&gt; [37] pkgconfig_2.0.3         desc_1.4.3              pkgdown_2.1.3          </span></span>
<span><span class="co">#&gt; [40] pillar_1.11.0           bslib_0.9.0             glue_1.8.0             </span></span>
<span><span class="co">#&gt; [43] Rcpp_1.1.0              harmonicmeanp_3.0.1     systemfonts_1.2.3      </span></span>
<span><span class="co">#&gt; [46] xfun_0.52               tibble_3.3.0            tidyselect_1.2.1       </span></span>
<span><span class="co">#&gt; [49] knitr_1.50              htmltools_0.5.8.1       rmarkdown_2.29         </span></span>
<span><span class="co">#&gt; [52] compiler_4.5.1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Julian Stamp, Lorin Crawford.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
