<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>How To Create a Mask File • sme</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="How To Create a Mask File">
<script defer data-domain="pkgdown.r-lib.org,all.tidyverse.org" src="https://plausible.io/js/plausible.js"></script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">sme</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/sme.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Tutorials</h6></li>
    <li><a class="dropdown-item" href="../articles/tutorial-cite-us.html">How To Cite Our Work</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-create-mask-file.html">How To Create a Mask File</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-memory-optimization.html">How To Optimize the Memory Requirements of SME</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-simulate-traits.html">How To Simulate Traits</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Studies</h6></li>
    <li><a class="dropdown-item" href="../articles/study-erythroid-differentiation-data.html">Conditioning Epistasis Search on Open Chromatin</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/lcrawlab/sme/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>How To Create a Mask File</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lcrawlab/sme/blob/v0.0.1/vignettes/tutorial-create-mask-file.Rmd" class="external-link"><code>vignettes/tutorial-create-mask-file.Rmd</code></a></small>
      <div class="d-none name"><code>tutorial-create-mask-file.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lcrawlab/sme" class="external-link">sme</a></span><span class="op">)</span></span></code></pre></div>
<p>The Sparse Marginal Epistasis test (SME) requires a HDF5 file
containing binary masking data. The effectiveness of the test relies on
how informative the mask is for a specific trait. For instance, to test
for marginal epistasis related to transcriptional activity, DNase I
hypersensitivity sites data can be used to condition the detection on
accessible chromatin regions.</p>
<p><img src="../reference/figures/sme-schematic-dhs.png" width="100%" alt="Sparse Marginal Epistasis test (SME) schematic DHS data"><strong>Figure 1.</strong> The illustration shows open chromatin data.
DNase I hypersensitive sites indicate transcriptional activity. The
DNase-seq signal is converted to a binary mask, where genetic variants
in inaccessible chromatin regions are excluded from the covariance
matrix for marginal epistasis.</p>
<div class="section level2">
<h2 id="mask-file-format">Mask File Format<a class="anchor" aria-label="anchor" href="#mask-file-format"></a>
</h2>
<p>The <code><a href="../reference/sme.html">sme()</a></code> function expects the mask data to be in an
HDF5 file. The HDF5 format includes two primary object types:</p>
<ul>
<li>Datasets - typed multidimensional arrays</li>
<li>Groups - container structures that can hold datasets and other
groups</li>
</ul>
</div>
<div class="section level2">
<h2 id="mask-format-requirements">Mask Format Requirements<a class="anchor" aria-label="anchor" href="#mask-format-requirements"></a>
</h2>
<p>The mask data should be organized into the following groups and
datasets:</p>
<p><strong>Groups</strong>:</p>
<ul>
<li>
<code>ld</code>: Contains SNPs in linkage disequilibrium (LD) with
the focal SNP, which will be <strong>excluded</strong>.</li>
<li>
<code>gxg</code>: Contains indices of SNPs used to condition the
marginal epistasis test, which will be <strong>included</strong>.</li>
</ul>
<p>The required group names can be configured as input parameters of
<code><a href="../reference/sme.html">sme()</a></code>. The defaults are <code>ld</code> and
<code>gxg</code>.</p>
<p><strong>Datasets</strong>:</p>
<ul>
<li>
<code>ld/&lt;j&gt;</code>: For each focal SNP
<code>&lt;j&gt;</code>, this dataset contains indices of SNPs in the
same LD block as that SNP. These SNPs will be <strong>excluded</strong>
from the gene-by-gene interaction covariance matrix.</li>
<li>
<code>gxg/&lt;j&gt;</code>: For each focal SNP
<code>&lt;j&gt;</code>, this dataset contains indices of SNPs to
<strong>include</strong> in the the gene-by-gene interaction covariance
matrix for focal SNP <code>&lt;j&gt;</code>.</li>
</ul>
<p><strong>Important</strong>: All indices in the mask file must be
<strong>zero-based</strong> and should correspond to the zero-based row
indices of the PLINK <code>.bim</code> file. This includes the dataset
index (<code>&lt;j&gt;</code> in <code>gxg/&lt;j&gt;</code>) and the
data itself. This zero-based indexing is necessary because the mask data
is read by a C++ subroutine in <code><a href="../reference/sme.html">sme()</a></code>, which uses zero-based
indexing, unlike R’s one-based indexing for SNP indices in the function
call.</p>
</div>
<div class="section level2">
<h2 id="creating-and-using-mask-files">Creating and Using Mask Files<a class="anchor" aria-label="anchor" href="#creating-and-using-mask-files"></a>
</h2>
<p>The package provides utility functions to create, write, and read
valid mask files for <code><a href="../reference/sme.html">sme()</a></code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hdf5_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Group names</span></span>
<span><span class="va">gxg_h5_group</span> <span class="op">&lt;-</span> <span class="st">"gxg"</span></span>
<span><span class="va">ld_h5_group</span> <span class="op">&lt;-</span> <span class="st">"ld"</span></span>
<span></span>
<span><span class="co"># Data (still in 1-based R indexing)</span></span>
<span><span class="va">include_gxg_snps</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span></span>
<span><span class="va">exclude_ld_snps</span> <span class="op">&lt;-</span> <span class="fl">5</span><span class="op">:</span><span class="fl">6</span></span>
<span></span>
<span><span class="co"># Focal SNP (still in 1-based R indexing)</span></span>
<span><span class="va">focal_snp</span> <span class="op">&lt;-</span> <span class="fl">4</span></span>
<span></span>
<span><span class="co"># Dataset names</span></span>
<span><span class="va">dataset_name_pattern</span> <span class="op">&lt;-</span> <span class="st">"%s/%s"</span></span>
<span><span class="co"># 0-based index!</span></span>
<span><span class="va">gxg_dataset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="va">dataset_name_pattern</span>, <span class="va">gxg_h5_group</span>, <span class="va">focal_snp</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">ld_dataset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="va">dataset_name_pattern</span>, <span class="va">ld_h5_group</span>, <span class="va">focal_snp</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create an empty HDF5 file</span></span>
<span><span class="fu"><a href="../reference/create_hdf5_file.html">create_hdf5_file</a></span><span class="op">(</span><span class="va">hdf5_file</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Write LD data</span></span>
<span><span class="fu"><a href="../reference/write_hdf5_dataset.html">write_hdf5_dataset</a></span><span class="op">(</span><span class="va">hdf5_file</span>, <span class="va">ld_dataset</span>, <span class="va">exclude_ld_snps</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span> <span class="co"># 0-based index!</span></span>
<span></span>
<span><span class="co"># Write GXG data</span></span>
<span><span class="fu"><a href="../reference/write_hdf5_dataset.html">write_hdf5_dataset</a></span><span class="op">(</span><span class="va">hdf5_file</span>, <span class="va">gxg_dataset</span>, <span class="va">include_gxg_snps</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>We can check that the data was written correctly.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ld_read</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_hdf5_dataset.html">read_hdf5_dataset</a></span><span class="op">(</span><span class="va">hdf5_file</span>, <span class="va">ld_dataset</span><span class="op">)</span></span>
<span><span class="va">gxg_read</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_hdf5_dataset.html">read_hdf5_dataset</a></span><span class="op">(</span><span class="va">hdf5_file</span>, <span class="va">gxg_dataset</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"Zero-based indices of SNPs to exclude: %s"</span>, <span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">ld_read</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  int [1:2] 4 5</span></span>
<span><span class="co">#&gt; character(0)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"Zero-based indices of SNPs to include: %s"</span>, <span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">gxg_read</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  int [1:10] 0 1 2 3 4 5 6 7 8 9</span></span>
<span><span class="co">#&gt; character(0)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] sme_0.0.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] jsonlite_1.8.9      dplyr_1.1.4         compiler_4.4.2     </span></span>
<span><span class="co">#&gt;  [4] tidyselect_1.2.1    Rcpp_1.0.13-1       FMStable_0.1-4     </span></span>
<span><span class="co">#&gt;  [7] parallel_4.4.2      tidyr_1.3.1         jquerylib_0.1.4    </span></span>
<span><span class="co">#&gt; [10] systemfonts_1.1.0   textshaping_0.4.1   harmonicmeanp_3.0.1</span></span>
<span><span class="co">#&gt; [13] yaml_2.3.10         fastmap_1.2.0       R6_2.5.1           </span></span>
<span><span class="co">#&gt; [16] generics_0.1.3      RcppEigen_0.3.4.0.2 knitr_1.49         </span></span>
<span><span class="co">#&gt; [19] genio_1.1.2         iterators_1.0.14    backports_1.5.0    </span></span>
<span><span class="co">#&gt; [22] checkmate_2.3.2     tibble_3.2.1        desc_1.4.3         </span></span>
<span><span class="co">#&gt; [25] bslib_0.8.0         pillar_1.10.1       rlang_1.1.4        </span></span>
<span><span class="co">#&gt; [28] cachem_1.1.0        xfun_0.50           fs_1.6.5           </span></span>
<span><span class="co">#&gt; [31] sass_0.4.9          cli_3.6.3           pkgdown_2.1.1      </span></span>
<span><span class="co">#&gt; [34] magrittr_2.0.3      digest_0.6.37       mvMAPIT_2.0.3      </span></span>
<span><span class="co">#&gt; [37] foreach_1.5.2       mvtnorm_1.3-3       lifecycle_1.0.4    </span></span>
<span><span class="co">#&gt; [40] CompQuadForm_1.4.3  vctrs_0.6.5         evaluate_1.0.3     </span></span>
<span><span class="co">#&gt; [43] glue_1.8.0          codetools_0.2-20    ragg_1.3.3         </span></span>
<span><span class="co">#&gt; [46] purrr_1.0.2         rmarkdown_2.29      tools_4.4.2        </span></span>
<span><span class="co">#&gt; [49] pkgconfig_2.0.3     htmltools_0.5.8.1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Julian Stamp, Lorin Crawford.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
