<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>How To Optimize the Memory Requirements of SME • sme</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="How To Optimize the Memory Requirements of SME">
<script defer data-domain="pkgdown.r-lib.org,all.tidyverse.org" src="https://plausible.io/js/plausible.js"></script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">sme</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/sme.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Tutorials</h6></li>
    <li><a class="dropdown-item" href="../articles/tutorial-cite-us.html">How To Cite Our Work</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-create-mask-file.html">How To Create a Mask File</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-memory-optimization.html">How To Optimize the Memory Requirements of SME</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-simulate-traits.html">How To Simulate Traits</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Studies</h6></li>
    <li><a class="dropdown-item" href="../articles/study-erythroid-differentiation-data.html">Conditioning Epistasis Search on Open Chromatin</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/lcrawlab/sme/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>How To Optimize the Memory Requirements of SME</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lcrawlab/sme/blob/main/vignettes/tutorial-memory-optimization.Rmd" class="external-link"><code>vignettes/tutorial-memory-optimization.Rmd</code></a></small>
      <div class="d-none name"><code>tutorial-memory-optimization.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lcrawlab/sme" class="external-link">sme</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/knitr/" class="external-link">knitr</a></span><span class="op">)</span></span></code></pre></div>
<p>This tutorial outlines the factors that affect memory usage when
running the <code><a href="../reference/sme.html">sme()</a></code> function and how to adjust parameters to
optimize memory consumption based on your available resources. To
estimate memory requirements, use the
<code><a href="../reference/approximate_memory_requirements.html">approximate_memory_requirements()</a></code> function, which provides
a rough estimate of the C++ objects stored in memory during the
<code><a href="../reference/sme.html">sme()</a></code> function call, though it may not reflect exact memory
usage.</p>
<div class="section level2">
<h2 id="genotype-data-size-and-number-of-blocks">Genotype Data Size and Number of Blocks<a class="anchor" aria-label="anchor" href="#genotype-data-size-and-number-of-blocks"></a>
</h2>
<p>The sample size is the primary factor influencing memory
requirements. Both phenotype and genotype data need to be loaded into
memory for computation. For large datasets, like Biobank-scale data
(350k samples and 500k SNPs), loading the entire dataset into memory
requires about 1.4TB (assuming double precision for the data matrix),
which exceeds most machines’ capacities.</p>
<p>To manage large datasets more efficiently, <code><a href="../reference/sme.html">sme()</a></code> reads
the genotype data in smaller blocks. The parameter <code>n_blocks</code>
controls the number of blocks. For instance, with 500k SNPs, setting
<code>n_blocks = 100</code> will load 5000 SNPs into memory at a time,
reducing the memory load and allowing computations to proceed block by
block.</p>
</div>
<div class="section level2">
<h2 id="number-of-random-vectors">Number of Random Vectors<a class="anchor" aria-label="anchor" href="#number-of-random-vectors"></a>
</h2>
<p>The <code><a href="../reference/sme.html">sme()</a></code> function uses a stochastic trace estimator to
approximate the trace of matrix products efficiently. The number of
random vectors impacts both the accuracy of trace estimates and memory
and computational efficiency.</p>
<p>During blockwise computation, the algorithm stores intermediate
matrices sized <code>sample_size x n_randvecs</code>. Increasing the
number of random vectors improves accuracy but also increases memory
usage and computation time. Typically, using around 10 random vectors
provides reasonably accurate results.</p>
</div>
<div class="section level2">
<h2 id="number-of-snps-sharing-random-vectors">Number of SNPs Sharing Random Vectors<a class="anchor" aria-label="anchor" href="#number-of-snps-sharing-random-vectors"></a>
</h2>
<p>The <code>chunk_size</code> parameter controls how many SNPs share
the same set of random vectors, enhancing the efficiency of genome-wide
data processing. This method reduces redundant calculations of the
genetic relatedness covariance matrix and minimizes the time spent
reading genotype data into memory.</p>
<p>For each set of SNPs analyzed together (in a “chunk”), intermediate
results must be stored. Consequently, the memory requirement grows with
the chunk size, calculated as:
<code>chunk_size x (sample_size x n_randvecs)</code>.</p>
<p><img src="../reference/figures/sme-shared-rvs.png" width="100%" alt="Sparse Marginal Epistasis test (SME) schematic pseudo-random vectors"><strong>Figure 1.</strong> Schematic overview illustrating the
compuational speedup resulting from sharing random vectors.
<strong>(a)</strong> In the randomized trace estimates we can identify
reusable matrix by vector products. Computing the exact trace of a
product of two covariance matrices is prohibitively computationally
expensive. Instead, the sparse marginal epistasis (SME) test
approximates the traces using random vectors
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>z</mi><annotation encoding="application/x-tex">z</annotation></semantics></math>.
For the full MQS computation of the point estimates of the variance
components, we see that the matrix-by-vector products of the form
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>A</mi><mi>z</mi></mrow><annotation encoding="application/x-tex">Az</annotation></semantics></math>
with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>A</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mi>K</mi><mo>,</mo><mi>G</mi><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">A \in \{K, G\}</annotation></semantics></math>
appear repeatedly. <strong>(b)</strong> The genetic relatedness matrix
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>K</mi><annotation encoding="application/x-tex">K</annotation></semantics></math>
is the same for all focal SNPs. Using unique random vectors in this
computation for every focal SNP, we compute the same quantity
repeatedly. Computing the matrix-by-vector products
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>K</mi><mi>z</mi></mrow><annotation encoding="application/x-tex">Kz</annotation></semantics></math>
constitutes almost half of the computation time of the point estimates.
<strong>(c)</strong> By sharing random vectors
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>z</mi><annotation encoding="application/x-tex">z</annotation></semantics></math>
between focal SNPs, computing
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>K</mi><mi>z</mi></mrow><annotation encoding="application/x-tex">Kz</annotation></semantics></math>
can be done once for all focal SNPs that share random vectors. With
this, the computation time of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>K</mi><mi>z</mi></mrow><annotation encoding="application/x-tex">Kz</annotation></semantics></math>
becomes negligible.</p>
</div>
<div class="section level2">
<h2 id="genotype-masking-for-the-gene-by-gene-interaction-covariance">Genotype Masking for the Gene-by-Gene Interaction Covariance<a class="anchor" aria-label="anchor" href="#genotype-masking-for-the-gene-by-gene-interaction-covariance"></a>
</h2>
<p>Masking genotypes that do not contribute to epistasis can help reduce
memory usage and computation time. When masked, these genotypes do not
need to be stored in memory, significantly decreasing memory
requirements. Note that the
<code><a href="../reference/approximate_memory_requirements.html">approximate_memory_requirements()</a></code> function does not account
for this reduction.</p>
</div>
<div class="section level2">
<h2 id="explore-the-memory-requirements">Explore the Memory Requirements<a class="anchor" aria-label="anchor" href="#explore-the-memory-requirements"></a>
</h2>
<p>To estimate memory needs based on your chosen parameters, use the
<code><a href="../reference/approximate_memory_requirements.html">approximate_memory_requirements()</a></code> function. This function
helps you determine if your planned settings will fit within available
memory and identify which parameters can be adjusted to meet your
resource constraints. The parameters <code>n_blocks</code>,
<code>n_randvecs</code>, and <code>chunk_size</code> are particularly
flexible and have a significant impact on memory usage. Note however,
that it does not account for masking and therefore likely overestimates
the required memory.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">350000</span><span class="op">)</span></span>
<span><span class="va">n_snps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">500000</span><span class="op">)</span></span>
<span><span class="va">n_blocks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">100</span>, <span class="fl">1000</span><span class="op">)</span></span>
<span><span class="va">n_randvecs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">chunk_size</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="va">parameters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/expand.html" class="external-link">crossing</a></span><span class="op">(</span></span>
<span>  n_samples <span class="op">=</span> <span class="va">n_samples</span>,</span>
<span>  n_snps <span class="op">=</span> <span class="va">n_snps</span>,</span>
<span>  n_blocks <span class="op">=</span> <span class="va">n_blocks</span>,</span>
<span>  n_randvecs <span class="op">=</span> <span class="va">n_randvecs</span>,</span>
<span>  chunk_size <span class="op">=</span> <span class="va">chunk_size</span></span>
<span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">estimated_memory</span> <span class="op">&lt;-</span> <span class="va">parameters</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>memory_gb <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="../reference/approximate_memory_requirements.html">approximate_memory_requirements</a></span><span class="op">(</span><span class="va">n_samples</span>,</span>
<span>                                    <span class="va">n_snps</span>,</span>
<span>                                    <span class="va">n_blocks</span>,</span>
<span>                                    <span class="va">n_randvecs</span>,</span>
<span>                                    <span class="va">chunk_size</span><span class="op">)</span>,</span>
<span>    <span class="fl">2</span></span>
<span>  <span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">estimated_memory</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">n_samples</th>
<th align="right">n_snps</th>
<th align="right">n_blocks</th>
<th align="right">n_randvecs</th>
<th align="right">chunk_size</th>
<th align="right">memory_gb</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">350000</td>
<td align="right">5e+05</td>
<td align="right">1</td>
<td align="right">10</td>
<td align="right">10</td>
<td align="right">797.64</td>
</tr>
<tr class="even">
<td align="right">350000</td>
<td align="right">5e+05</td>
<td align="right">1</td>
<td align="right">10</td>
<td align="right">100</td>
<td align="right">7322.88</td>
</tr>
<tr class="odd">
<td align="right">350000</td>
<td align="right">5e+05</td>
<td align="right">1</td>
<td align="right">100</td>
<td align="right">10</td>
<td align="right">801.16</td>
</tr>
<tr class="even">
<td align="right">350000</td>
<td align="right">5e+05</td>
<td align="right">1</td>
<td align="right">100</td>
<td align="right">100</td>
<td align="right">7347.53</td>
</tr>
<tr class="odd">
<td align="right">350000</td>
<td align="right">5e+05</td>
<td align="right">100</td>
<td align="right">10</td>
<td align="right">10</td>
<td align="right">8.73</td>
</tr>
<tr class="even">
<td align="right">350000</td>
<td align="right">5e+05</td>
<td align="right">100</td>
<td align="right">10</td>
<td align="right">100</td>
<td align="right">79.24</td>
</tr>
<tr class="odd">
<td align="right">350000</td>
<td align="right">5e+05</td>
<td align="right">100</td>
<td align="right">100</td>
<td align="right">10</td>
<td align="right">12.25</td>
</tr>
<tr class="even">
<td align="right">350000</td>
<td align="right">5e+05</td>
<td align="right">100</td>
<td align="right">100</td>
<td align="right">100</td>
<td align="right">103.89</td>
</tr>
<tr class="odd">
<td align="right">350000</td>
<td align="right">5e+05</td>
<td align="right">1000</td>
<td align="right">10</td>
<td align="right">10</td>
<td align="right">1.56</td>
</tr>
<tr class="even">
<td align="right">350000</td>
<td align="right">5e+05</td>
<td align="right">1000</td>
<td align="right">10</td>
<td align="right">100</td>
<td align="right">13.39</td>
</tr>
<tr class="odd">
<td align="right">350000</td>
<td align="right">5e+05</td>
<td align="right">1000</td>
<td align="right">100</td>
<td align="right">10</td>
<td align="right">5.08</td>
</tr>
<tr class="even">
<td align="right">350000</td>
<td align="right">5e+05</td>
<td align="right">1000</td>
<td align="right">100</td>
<td align="right">100</td>
<td align="right">38.04</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="a-note-on-the-runtime-of-sme">A Note on the Runtime of SME<a class="anchor" aria-label="anchor" href="#a-note-on-the-runtime-of-sme"></a>
</h2>
<p>Despite the computational efficiency of SME, genome-wide testing
requires considerable resources. We recommend to analyze data in
batches, and to launch multiple processes simultaneously on a
high-performance cluster (HPC). In <a href="articles/study-erythroid-differentiation-data.html">this
study</a>, we analyzed 544k SNPs genotype in 350k individuals. We
launched 544 slurm jobs requesting 43GB memory and 6 CPUs each to
analyze batches of 1000 SNPs with chunk sizes of 250 SNPs. Genome-wide
testing of a single trait on an HPC with 960 CPUs and 6840GB of memory
available took about 3.5 days.</p>
<p><img src="../reference/figures/sme-runtime.png" width="100%" alt="Sparse Marginal Epistasis test (SME) runtime"><strong>Figure 2. </strong> SME has improved power to detect marginal
epistasis and runs 10x to 90x faster than state-of-the-art methods. The
CPU time was measured on 350,000 individuals.</p>
</div>
<div class="section level2">
<h2 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] knitr_1.49  dplyr_1.1.4 tidyr_1.3.1 sme_0.0.1  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] jsonlite_1.8.9      compiler_4.4.2      tidyselect_1.2.1   </span></span>
<span><span class="co">#&gt;  [4] Rcpp_1.0.14         FMStable_0.1-4      parallel_4.4.2     </span></span>
<span><span class="co">#&gt;  [7] jquerylib_0.1.4     systemfonts_1.1.0   textshaping_0.4.1  </span></span>
<span><span class="co">#&gt; [10] harmonicmeanp_3.0.1 yaml_2.3.10         fastmap_1.2.0      </span></span>
<span><span class="co">#&gt; [13] R6_2.5.1            generics_0.1.3      RcppEigen_0.3.4.0.2</span></span>
<span><span class="co">#&gt; [16] genio_1.1.2         iterators_1.0.14    backports_1.5.0    </span></span>
<span><span class="co">#&gt; [19] checkmate_2.3.2     tibble_3.2.1        desc_1.4.3         </span></span>
<span><span class="co">#&gt; [22] bslib_0.8.0         pillar_1.10.1       rlang_1.1.4        </span></span>
<span><span class="co">#&gt; [25] cachem_1.1.0        xfun_0.50           fs_1.6.5           </span></span>
<span><span class="co">#&gt; [28] sass_0.4.9          cli_3.6.3           pkgdown_2.1.1      </span></span>
<span><span class="co">#&gt; [31] magrittr_2.0.3      digest_0.6.37       mvMAPIT_2.0.3      </span></span>
<span><span class="co">#&gt; [34] foreach_1.5.2       mvtnorm_1.3-3       lifecycle_1.0.4    </span></span>
<span><span class="co">#&gt; [37] CompQuadForm_1.4.3  vctrs_0.6.5         evaluate_1.0.3     </span></span>
<span><span class="co">#&gt; [40] glue_1.8.0          codetools_0.2-20    ragg_1.3.3         </span></span>
<span><span class="co">#&gt; [43] purrr_1.0.2         rmarkdown_2.29      tools_4.4.2        </span></span>
<span><span class="co">#&gt; [46] pkgconfig_2.0.3     htmltools_0.5.8.1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Julian Stamp, Lorin Crawford.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
