<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>How To Simulate Traits • smer</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="How To Simulate Traits">
<script defer data-domain="pkgdown.r-lib.org,all.tidyverse.org" src="https://plausible.io/js/plausible.js"></script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">smer</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/smer.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Tutorials</h6></li>
    <li><a class="dropdown-item" href="../articles/tutorial-cite-us.html">How To Cite Our Work</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-create-mask-file.html">How To Create a Mask File</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-memory-optimization.html">How To Optimize the Memory Requirements of SME</a></li>
    <li><a class="dropdown-item" href="../articles/tutorial-simulate-traits.html">How To Simulate Traits</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Studies</h6></li>
    <li><a class="dropdown-item" href="../articles/study-erythroid-differentiation-data.html">Conditioning Epistasis Search on Open Chromatin</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/lcrawlab/sme/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>How To Simulate Traits</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lcrawlab/sme/blob/main/vignettes/tutorial-simulate-traits.Rmd" class="external-link"><code>vignettes/tutorial-simulate-traits.Rmd</code></a></small>
      <div class="d-none name"><code>tutorial-simulate-traits.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lcrawlab/sme" class="external-link">smer</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/OchoaLab/genio" class="external-link">genio</a></span><span class="op">)</span></span></code></pre></div>
<p>For a simple illustration, we generate a synthetic allele count
matrix with minor allele frequency <code>maf</code> greater than 5%. The
function <code><a href="../reference/simulate_traits.html">simulate_traits()</a></code> requires the input genotype data
to be stored in the PLINK format (<code>.bim</code>, <code>.bed</code>,
and <code>.fam</code> files). Here we use the package <code>genio</code>
to write the allele count matrix to a PLINK file.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plink_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">n_samples</span> <span class="op">&lt;-</span> <span class="fl">3000</span></span>
<span><span class="va">n_snps</span> <span class="op">&lt;-</span> <span class="fl">5000</span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fl">0.05</span> <span class="op">+</span> <span class="fl">0.45</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">n_snps</span><span class="op">)</span></span>
<span><span class="va">random_minor_allele_counts</span>   <span class="op">&lt;-</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">n_samples</span> <span class="op">*</span> <span class="va">n_snps</span><span class="op">)</span> <span class="op">&lt;</span> <span class="va">maf</span><span class="op">)</span> <span class="op">+</span></span>
<span>                                <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">n_samples</span> <span class="op">*</span> <span class="va">n_snps</span><span class="op">)</span> <span class="op">&lt;</span> <span class="va">maf</span><span class="op">)</span></span>
<span><span class="va">allele_count_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="va">random_minor_allele_counts</span>,</span>
<span>  nrow <span class="op">=</span> <span class="va">n_samples</span>,</span>
<span>  ncol <span class="op">=</span> <span class="va">n_snps</span>,</span>
<span>  byrow <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create .fam and .bim data frames</span></span>
<span><span class="va">fam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  fam <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"F%d"</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span><span class="op">)</span>,       <span class="co"># Family ID</span></span>
<span>  id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"I%d"</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span><span class="op">)</span>,       <span class="co"># Individual ID</span></span>
<span>  pat <span class="op">=</span> <span class="fl">0</span>,                        <span class="co"># Paternal ID</span></span>
<span>  mat <span class="op">=</span> <span class="fl">0</span>,                        <span class="co"># Maternal ID</span></span>
<span>  sex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="va">n_samples</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  pheno <span class="op">=</span> <span class="op">-</span><span class="fl">9</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">bim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  chr <span class="op">=</span> <span class="fl">1</span>,               <span class="co"># Chromosome number</span></span>
<span>  id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"rs%d"</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_snps</span><span class="op">)</span>,   <span class="co"># SNP name</span></span>
<span>  posg <span class="op">=</span> <span class="fl">0</span>,                         <span class="co"># Genetic distance (cM)</span></span>
<span>  pos <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">n_snps</span>,          <span class="co"># Base-pair position</span></span>
<span>  ref <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"C"</span>, <span class="st">"G"</span>, <span class="st">"T"</span><span class="op">)</span>, <span class="va">n_snps</span>, replace <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>, <span class="co"># Minor allele</span></span>
<span>  alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"C"</span>, <span class="st">"G"</span>, <span class="st">"T"</span><span class="op">)</span>, <span class="va">n_snps</span>, replace <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>  <span class="co"># Major allele</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Write to .bed, .fam, and .bim files</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/genio/man/write_plink.html" class="external-link">write_plink</a></span><span class="op">(</span></span>
<span>  file <span class="op">=</span> <span class="va">plink_file</span>,</span>
<span>  X <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">allele_count_matrix</span><span class="op">)</span>,</span>
<span>  fam <span class="op">=</span> <span class="va">fam</span>,</span>
<span>  bim <span class="op">=</span> <span class="va">bim</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>To simulate traits, we need to specify the genetic architecture.
E.g., the function requires us to specify</p>
<ul>
<li>the narrow sense heritability
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>h</mi><mn>2</mn></msup><annotation encoding="application/x-tex">h^2</annotation></semantics></math>
</li>
<li>the indices SNPs contributing to the narrow sense heritability</li>
<li>the heritability due to epistasis</li>
<li>the indices of SNPs contributing to the epistatic trait
variance</li>
<li>the path to the output file.</li>
</ul>
<p>The simulated trait is written to file in the PLINK phenotype
format.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">additive_heritability</span> <span class="op">&lt;-</span> <span class="fl">0.3</span></span>
<span><span class="va">gxg_heritability</span> <span class="op">&lt;-</span> <span class="fl">0.25</span></span>
<span><span class="va">n_snps_gxg_group</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">n_snps_additive</span> <span class="op">&lt;-</span> <span class="fl">100</span></span>
<span><span class="va">additive_snps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="va">n_snps</span>, <span class="va">n_snps_additive</span>, replace <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gxg_group_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">additive_snps</span>, <span class="va">n_snps_gxg_group</span>, replace <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gxg_group_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">additive_snps</span>, <span class="va">gxg_group_1</span><span class="op">)</span>,</span>
<span>                           <span class="va">n_snps_gxg_group</span>, replace <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/simulate_traits.html">simulate_traits</a></span><span class="op">(</span></span>
<span>  <span class="va">plink_file</span>,</span>
<span>  <span class="va">pheno_file</span>,</span>
<span>  <span class="va">additive_heritability</span>,</span>
<span>  <span class="va">gxg_heritability</span>,</span>
<span>  <span class="va">additive_snps</span>,</span>
<span>  <span class="va">gxg_group_1</span>,</span>
<span>  <span class="va">gxg_group_2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The variance components of the traits are constructed to match the
input. See Crawford et al. (2017) and <a href="https://lcrawlab.github.io/mvMAPIT/articles/tutorial-simulations.html" class="external-link">mvMAPIT
Documentation: Simulate Traits</a> for a full description of the
simulation scheme.</p>
<div class="section level2">
<h2 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] genio_1.1.2 smer_0.0.2 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] jsonlite_2.0.0      dplyr_1.1.4         compiler_4.5.1     </span></span>
<span><span class="co">#&gt;  [4] tidyselect_1.2.1    Rcpp_1.1.0          FMStable_0.1-4     </span></span>
<span><span class="co">#&gt;  [7] parallel_4.5.1      tidyr_1.3.1         jquerylib_0.1.4    </span></span>
<span><span class="co">#&gt; [10] systemfonts_1.2.3   textshaping_1.0.1   harmonicmeanp_3.0.1</span></span>
<span><span class="co">#&gt; [13] yaml_2.3.10         fastmap_1.2.0       R6_2.6.1           </span></span>
<span><span class="co">#&gt; [16] generics_0.1.4      knitr_1.50          iterators_1.0.14   </span></span>
<span><span class="co">#&gt; [19] backports_1.5.0     checkmate_2.3.3     tibble_3.3.0       </span></span>
<span><span class="co">#&gt; [22] desc_1.4.3          bslib_0.9.0         pillar_1.11.0      </span></span>
<span><span class="co">#&gt; [25] rlang_1.1.6         cachem_1.1.0        xfun_0.53          </span></span>
<span><span class="co">#&gt; [28] fs_1.6.6            sass_0.4.10         cli_3.6.5          </span></span>
<span><span class="co">#&gt; [31] pkgdown_2.1.3       magrittr_2.0.3      digest_0.6.37      </span></span>
<span><span class="co">#&gt; [34] mvMAPIT_2.0.3       foreach_1.5.2       mvtnorm_1.3-3      </span></span>
<span><span class="co">#&gt; [37] lifecycle_1.0.4     CompQuadForm_1.4.4  vctrs_0.6.5        </span></span>
<span><span class="co">#&gt; [40] evaluate_1.0.4      glue_1.8.0          codetools_0.2-20   </span></span>
<span><span class="co">#&gt; [43] ragg_1.4.0          purrr_1.1.0         rmarkdown_2.29     </span></span>
<span><span class="co">#&gt; [46] tools_4.5.1         pkgconfig_2.0.3     htmltools_0.5.8.1</span></span></code></pre></div>
</div>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Julian Stamp, Lorin Crawford.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
